### ISSUE TITLE
CPU eval BatchNorm2d is not threaded

### ISSUE THREAD

**Issue by jamesr66a** [pytorch/pytorch#52011]
```
import torch
import torch.fx
import torchvision.models as models

rn18 = models.resnet18()
# comment this out to do train mode
rn18.eval()

inp = torch.randn(5, 3, 224, 224)
bn_input = rn18.conv1(inp)

NITER = 1000
import time
s = time.time()
for _ in range(NITER):
    out = rn18.bn1(bn_input)
e = time.time()

time_per_iter = (e - s) / NITER
bytes_rw = bn_input.numel() * bn_input.element_size() + out.numel() * bn_input.element_size()
print('GB/s', bytes_rw / time_per_iter / 1e9)
"""
OMP_NUM_THREADS=default
eval: GB/s 5.64837339732215
train: GB/s 24.822818017881474

OMP_NUM_THREADS=1
eval: GB/s 5.809292354256564
train: GB/s 1.9974463507138427
"""
```

With intra-op parallelism, BatchNorm2d in train mode hits memory bandwidth bound. Eval does not, sitting at around 5.64 GB/s. With single threading, train's performance is worse, as expected. Eval's is the same, because it doesn't have threading. Please add threading to the eval version of BatchNorm2d

**Comment by bertmaher** [comment]
A funny tidbit here, is that I repro'ed this, and upon investigation found out that it's spending all its time in page faults:
![Screen Shot 2021-02-13 at 8 58 53 AM](https://user-images.githubusercontent.com/4441820/107855856-ce376600-6dd9-11eb-869e-5fafe4a18937.png)
This doesn't happen to me on every server so it's probably sensitive to the malloc library/kernel/something else.  And parallelization would help, it's just funny that it would be parallelizing the page fault handler more than anything :-D
